cmd/compareconfig.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/Azure/ARO-RP/pkg/deploy"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+func main() {
```

Comment:
nit: it might make sense to put this command in our `./hack` folder - as much as I don't like that becoming the kitchen sink, we may want to reserve the commands in `cmd` for our mission-critical ones. 

Not a huge deal either way though.